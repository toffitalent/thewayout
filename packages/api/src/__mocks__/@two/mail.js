const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
