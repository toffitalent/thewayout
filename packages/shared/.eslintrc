{
  "settings": {
    "import/resolver": {
      "typescript": {
        "project": ["tsconfig.json", "packages/shared/tsconfig.json"]
      }
    }
  },
  "rules": {
    "no-param-reassign": "off",
    "react/react-in-jsx-scope": "off"
  }
}
