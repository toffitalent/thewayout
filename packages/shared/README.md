# @two/shared

Copyright (c) 2023-Present The Way Out, Inc. All Rights Reserved.

## Development

This package contains code shared between the [api](../api/README.md), [app](../app/README.md), and [web](../web/README.md) packages and is not intended to run in a standalone manner. See the READMEs of those packages for information on starting a development environment.

## Testing

To run all component and unit tests and get coverage reports, simply run `yarn workspace @dlatwobs/shared test`.
