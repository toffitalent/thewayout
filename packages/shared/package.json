{
  "name": "@two/shared",
  "version": "1.0.0",
  "author": "Disruptive Labs",
  "license": "UNLICENSED",
  "private": true,
  "main": "build/index.js",
  "types": "build/index.d.ts",
  "scripts": {
    "build": "tsc -b tsconfig.build.json",
    "clean": "rimraf build coverage",
    "test": "jest --coverage",
    "type-check": "tsc"
  },
  "dependencies": {
    "@disruptive-labs/ui": "file:../../vendor/@disruptive-labs/ui",
    "axios": "^0.27.2",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "devDependencies": {
    "@testing-library/dom": "^9.3.1",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@testing-library/user-event": "^14.4.3",
    "@types/jest": "^29.5.2",
    "@types/react": "^18.2.14",
    "babel-jest": "^29.5.0",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "jest-environment-jsdom": "^29.5.0",
    "jest-junit": "^16.0.0",
    "react-test-renderer": "^18.2.0",
    "rimraf": "^3.0.2",
    "sass": "^1.63.6",
    "typescript": "^5.2.2"
  },
  "jest": {
    "clearMocks": true,
    "collectCoverageFrom": [
      "<rootDir>/src/**/*.{ts,tsx}"
    ],
    "globalSetup": "<rootDir>/test/setup.js",
    "moduleNameMapper": {
      "^.+\\.module\\.(css|sass|scss)$": "identity-obj-proxy",
      "^.+\\.(css|sass|scss)$": "<rootDir>/test/styleMock.js",
      "^.+\\.(jpg|jpeg|png|gif|webp|svg)$": "<rootDir>/test/fileMock.js",
      "^@test$": "<rootDir>/test/index.tsx"
    },
    "setupFilesAfterEnv": [
      "<rootDir>/test/jest.test.ts"
    ],
    "testEnvironment": "jsdom",
    "testPathIgnorePatterns": [
      "<rootDir>/node_modules/",
      "<rootDir>/test/"
    ],
    "transform": {
      "^.+\\.(js|jsx|ts|tsx)$": [
        "babel-jest",
        {
          "presets": [
            "next/babel"
          ]
        }
      ]
    },
    "transformIgnorePatterns": [
      "node_modules/(?!(@?disruptive-labs.*|@popperjs/core|es-cookie)/)"
    ]
  },
  "jest-junit": {
    "outputDirectory": "./artifacts/jest",
    "outputName": "results.xml"
  }
}
