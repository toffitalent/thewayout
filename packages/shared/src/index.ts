// NOTE: React components should not exported here

export * from './constants';
export * from './types';
export * from './utils';
