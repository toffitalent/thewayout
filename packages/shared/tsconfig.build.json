{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "composite": true,
    "incremental": true,
    "module": "commonjs",
    "noEmit": false,
    "outDir": "./build",
    "rootDir": "./src",
    "tsBuildInfoFile": "./build/.tsbuildinfo"
  },
  "include": ["**/*.ts", "**/*.tsx"],
  "exclude": ["build", "node_modules", "**/*.test.ts", "**/*.test.tsx", "test"]
}
