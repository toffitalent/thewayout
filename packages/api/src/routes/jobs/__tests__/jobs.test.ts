import { JobStatus } from '@two/shared';
import { auth, fixtures, request } from '@test';

describe('GET /api/v1/jobs/:jobId', () => {
  test('get a job as a client without application', async () => {
    const res = await request()
      .get(`/api/v1/jobs/${fixtures.job1.id}`)
      .use(auth(fixtures.user1.id, ['client']));

    expect(res.status).toBe(200);
    expect(res.body.applications).toBeUndefined();
  });

  test('get a job as a client with application', async () => {
    const res = await request()
      .get(`/api/v1/jobs/${fixtures.job1.id}`)
      .use(auth(fixtures.user4.id, ['client']));

    expect(res.status).toBe(200);
    expect(res.body.applications.length).toBe(1);
    expect(
      res.body.applications.filter(
        (application: any) => application.clientId === fixtures.clientProfile1.id,
      ).length,
    ).toBe(1);
  });

  test('get a job as a employer', async () => {
    const res = await request()
      .get(`/api/v1/jobs/${fixtures.job1.id}`)
      .use(auth(fixtures.user5.id, ['employer']));

    expect(res.status).toBe(200);
    expect(res.body.applications).toBeUndefined();
  });
});

describe('PATCH /api/v1/jobs/:jobId', () => {
  test('updates job', async () => {
    const res = await request()
      .patch(`/api/v1/jobs/${fixtures.job1.id}`)
      .send({
        status: JobStatus.paused,
      })
      .use(auth(fixtures.user6.id, ['employer']));

    expect(res.status).toBe(200);
    expect(res.body).toMatchSnapshot();
  });

  test('return error when its not employer job', async () => {
    const res = await request()
      .patch(`/api/v1/jobs/${fixtures.job1.id}`)
      .send({
        status: JobStatus.paused,
      })
      .use(auth(fixtures.user5.id, ['employer']));

    expect(res.status).toBe(403);
  });
});
