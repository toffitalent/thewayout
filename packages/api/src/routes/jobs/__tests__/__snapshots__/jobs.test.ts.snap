// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PATCH /api/v1/jobs/:jobId updates job 1`] = `
{
  "applicationsCount": 3,
  "createdAt": "<Date>",
  "department": "Customer Service",
  "description": "Delectus pariatur quae accusantium rerum consequatur. Veritatis id quisquam debitis temporibus quibusdam placeat ex saepe. Consequatur cum sint iusto sequi ducimus autem blanditiis libero. Itaque asperiores laudantium cumque. Ab molestias assumenda illum quas iste assumenda odit. Optio fuga reiciendis fuga.",
  "employer": {
    "address": "Wolff Spring",
    "availableJobsCount": 1,
    "availableProfilesUncloak": 3,
    "avatar": null,
    "city": "North Brooks",
    "createdAt": "<Date>",
    "description": "Eum eos laborum labore ipsam libero nobis. Earum maxime porro quas dicta vero. Occaecati impedit amet eaque doloribus. Veniam adipisci doloremque voluptas.",
    "id": "01817e24-0d42-763f-b7ea-8f975994f864",
    "industry": "Accounting",
    "name": "Oberbrunner, Dare and Trantow",
    "numberOfEmployees": "10000+",
    "postalCode": "04674",
    "state": "AK",
    "updatedAt": "<Date>",
    "yearsInBusiness": "0-3",
  },
  "experience": "entry",
  "hiredApplicationsCount": 0,
  "id": "01817e24-0d42-7b06-943b-43c05c9d4a6e",
  "location": null,
  "numberOfOpenPositions": 3,
  "offenses": [
    "distribution",
    "diuDwi",
    "battery",
    "burglary",
    "arson",
    "bailJumping",
  ],
  "offensesTypes": [
    "drugs",
    "motorVehicle",
    "propertyDamage",
    "theft",
    "violent",
    "whiteCollar",
  ],
  "questions": null,
  "responsibilities": [
    "Repudiandae iste ipsa officiis.
Molestias inventore sit blanditiis quibusdam rerum veniam dolorem nihil earum.
Aperiam a voluptates.
Corrupti voluptatum soluta ab at quaerat.
Delectus suscipit nesciunt aspernatur animi quisquam.",
    "Laborum ullam rerum modi.",
  ],
  "salaryOptions": {
    "bonuses": [],
    "description": "quo",
    "max": "40",
    "min": "30",
    "salary": "hourly",
  },
  "skillsDescription": "Ex vel minima. Aspernatur tenetur commodi beatae perferendis perferendis. Cum asperiores mollitia omnis excepturi tempore delectus. Voluptatem explicabo tempora necessitatibus eveniet.
Reiciendis id ratione. Ipsa ipsum et facilis aspernatur quisquam vitae officia rem. Maxime recusandae suscipit exercitationem nisi delectus atque. Placeat voluptates voluptate delectus cumque asperiores veniam maxime voluptates magnam. Est minima voluptatibus vel quod numquam. Magnam pariatur nam consectetur.
Dolore quisquam quisquam. Qui ad ab est. Ullam nostrum placeat. Tempora dignissimos id facere beatae aliquam doloremque voluptates. Omnis deleniti iusto iste asperiores dolorum ullam.",
  "startDate": "10.2022",
  "status": "paused",
  "title": "Central Quality Liaison",
  "typeOfWork": "hybrid",
  "updatedAt": "<Date>",
  "veteranOrJustice": [
    "justiceImpacted",
    "veteran",
  ],
  "workingTime": "fullTime",
}
`;
